plutus-core/untyped-plutus-core/test/Transform/Simplify.hs

```diff
@@ -226,6 +227,40 @@ multiApp = runQuote $ do
       app = mkIterAppNoAnn lam [mkConstant @Integer () 1, mkConstant @Integer () 2, mkConstant @Integer () 3]
   pure app
 
+-- | The UPLC term in this test should come from the following TPLC term after erasing its types:
+-- @ (/\(p :: *) -> \(x : p) -> /\(q :: *) -> \(y : q) -> /\(r :: *) -> \(z : r) -> z) Int 1 Int 2 Int 3 @
+forceApply :: Term Name PLC.DefaultUni PLC.DefaultFun ()
+forceApply = runQuote $ do
+  x <- freshName "x"
+  y <- freshName "y"
+  z <- freshName "z"
+  let one = mkConstant @Integer () 1
+      two = mkConstant @Integer () 2
+      three = mkConstant @Integer () 3
+      t = Delay () (LamAbs () x (Delay () (LamAbs () y (Delay () (LamAbs () z (Var () z))))))
+      app = Apply () (Force () (Apply () (Force () (Apply () (Force () t) one)) two)) three
+  pure app
+
+-- | The UPLC term in this test should come from the following TPLC term after erasing its types:
+-- @ (/\(p :: *) -> \(x : p) -> /\(q :: *) -> \(y1 : q) (y2 : String) (y3 : String) -> /\(r :: *) -> \(z : r) -> z) Int 1 Int 2 "foo" "bar" Int 3 @
+-- Note that compared to the 'forceApply' test above, this term has multiple term applications nested inside a single
+-- type abstraction/instantiation.
+forceMultipleApply :: Term Name PLC.DefaultUni PLC.DefaultFun ()
+forceMultipleApply = runQuote $ do
```

Comment:
WIP: this test isn't good enough because the term inside the body doesn't contain references to all the bound vars and the inlining optimisation simplifies the term